Add docs for changing cluster plan between Standard and Basic (#19462)
cockroachdb/docs: src/current/cockroachcloud/change-plan-between-basic-and-standard.md

---
title: Change a Cluster's Plan Between Basic and Standard
summary: Learn how to change a cluster's plan between a CockroachDB Basic cluster and a CockroachDB Standard cluster.
toc: true
---

{{site.data.alerts.callout_info}}
CockroachDB {{ site.data.products.standard }}, our new, enterprise-ready plan, is currently in [Preview]({% link {{ site.current_cloud_version }}/cockroachdb-feature-availability.md %}).
{{site.data.alerts.end}}

This page describes how to change a cluster's [plan]({% link cockroachcloud/index.md %}#plans) between CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} using the [{{ site.data.products.cloud }} Console](https://cockroachlabs.cloud/).

To use the CockroachDB Cloud API to change your cluster's plan, refer to [Use the CockroachDB Cloud API]({% link cockroachcloud/cloud-api.md %}#change-a-clusters-plan).

## Before you begin

> You'll need the [Cluster Administrator]({% link cockroachcloud/authorization.md %}#cluster-administrator) role on the running {{ site.data.products.standard }} or {{ site.data.products.basic }} cluster in order to change the cluster's [plan]({% link cockroachcloud/index.md %}#plans).

Reviewer: @katmayb Cluster Operator roles can also perform the plan change.

Author: done!

## Change a cluster's plan

{{site.data.alerts.callout_info}}
After changing your cluster's plan, you will not be able to change the plan again for 48 hours.
{{site.data.alerts.end}}

1. On the [**Clusters** page]({% link cockroachcloud/cluster-management.md %}#view-clusters-page), select the cluster that requires a plan change.
1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**.
1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be highlighted. Select **Next: Capacity**.
1. For capacity:

>     - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. Select **Next: Finalize**.

Reviewer: @katmayb we give a recommendation in CC console for your provisioned vCPU depending on your peak RU/s in the last 30 days. For your reference, we calculate this vCPU based on the 30-day peak RU/s plus a 40% buffer, then convert using 500 RU/s = 1 vCPU. We use a 10-minute roll-up for the peak RU/s. -- This calculation may be good to include in docs somewhere to reference; Mike had suggested adding it to "Plan your cluster".

There are also a few caveats that we may want to mention with respect to the recommendation in the UI: 
- This should be interpreted as a suggestion. Users should keep an eye on their CPU utilization % and adjust their compute capacity as needed. 
- Standard currently supports a maximum of 60 vCPUs. If we estimate you need more than 60 vCPU, we tell them that we estimate they need more than our capacity, but "You can provision up to 60 vCPU. If your workload consistently spikes above that level you may experience performance issues."

Author: I added a section to "plan your cluster" with the calculation. And I've added that the Cloud Console provides a suggestion but that users should monitor this.

Your last bullet point — I've added to the Plan your cluster page. I haven't not specifically mentioned the 60 vCPUs for ease of keeping the docs up to date. But, if that's a number that won't change frequently or you'd prefer it specified I can update these changes.

    - If you are changing to a [{{ site.data.products.basic }} plan]({% link cockroachcloud/create-a-basic-cluster.md %}#step-4-configure-cluster-capacity), choose whether the cluster should have **Unlimited** capacity to scale, or **Set a monthly limit**. Select **Next: Finalize**.

> 1. Review the changes and then **Update cluster**.

Reviewer: We include the following in the UI, if it makes sense to include here as well: 

- The upgrade process will take up to 5 mins
- Their cluster will remain functional under the current plan during the switching period
- They can switch back the plan if needed. But when you go from Standard to Basic, you need to wait 48 hours before being able to change the plan back to Standard.

Author: The 48 hours wait is just for if the cluster was originally Standard? I have a note earlier on the page that mentions this generally about any switch back, so maybe I need to change that?

I have added these as bullet points to the final step.

For changing plans from {{ site.data.products.standard }} to {{ site.data.products.basic }}, you must disable {{ site.data.products.standard }}-only features before starting the plan change.

### Considerations

- If the cluster was created on legacy architecture, then it cannot be upgraded in the Cloud Console. The cluster will not support multi-region features or private connections. To use these features, you must take a [backup]({% link cockroachcloud/backup-and-restore-overview.md %}) and restore into a new cluster. To change the plan for your current cluster despite the feature limitations, you can use the [Cloud API]({% link cockroachcloud/cloud-api.md %}#change-a-clusters-plan).
- The plan change will take up to 5 minutes.
- The cluster will remain operational under the current plan during the switching period.
- You can change back the plan if needed. However, when you change from {{ site.data.products.standard }} to {{ site.data.products.basic }}, you will need to wait 48 hours before being able to change back to {{ site.data.products.standard }}.

## Change a cluster's plan

{{site.data.alerts.callout_info}}

> After changing your cluster's plan, you will not be able to change the plan again for 48 hours.

Reviewer: This is only true for Standard to Basic, like you have it called out under considerations

Author: I have updated this to include the Standard to Basic callout.

{{site.data.alerts.end}}

1. On the [**Clusters** page]({% link cockroachcloud/cluster-management.md %}#view-clusters-page), select the cluster that requires a plan change.

> 1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**.

Author: ```suggestion
1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**. If this option is unavailable, refer to [Considerations](#considerations).
```

> 1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be highlighted. Select **Next: Capacity**.

Reviewer 2: <img width="807" alt="Screenshot 2025-04-03 at 2 46 10 PM" src="https://github.com/user-attachments/assets/2358c54b-e2f8-4aad-a548-f0348e3f6bf6" />

```suggestion
1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be labeled, while the other plan type will be highlighted. Select **Next: Capacity**.
```

1. For capacity:

>     - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. The Cloud Console provides a suggestion for provisioned vCPU, we recommend monitoring CPU utilization and adjusting the cluster's compute capacity as necessary. For additional information, refer to [Changing cluster plan provisioned capacity recommendations]({% link cockroachcloud/plan-your-cluster.md %}#changing-cluster-plan-provisioned-capacity-recommendations). Select **Next: Finalize**.

Author: ```suggestion
    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. The Cloud Console provides a suggestion for provisioned vCPU, we recommend monitoring CPU utilization and adjusting the cluster's compute capacity as necessary. For additional information, refer to [Recommended provisioned capacity when changing cluster plan]({% link cockroachcloud/plan-your-cluster.md %}#recommended-provisioned-capacity-when-changing-cluster-plan). Select **Next: Finalize**.
```

    - If you are changing to a [{{ site.data.products.basic }} plan]({% link cockroachcloud/create-a-basic-cluster.md %}#step-4-configure-cluster-capacity), choose whether the cluster should have **Unlimited** capacity to scale, or **Set a monthly limit**. Select **Next: Finalize**.

> 1. Review the changes and then **Update cluster**. Consider the following:

Reviewer 2: ```suggestion
1. Review the changes and then **Update cluster**.
```
Please either remove "consider the following" or add considerations.
